compiler/src/iree/compiler/Utils/ElementPackingUtils.cpp

```diff
@@ -15,7 +15,16 @@
 
 namespace mlir::iree_compiler {
 
+llvm::cl::opt<bool>
+    clEnableI1Support("iree-enable-i1",
+                      llvm::cl::desc("enable i1 data type codegen"),
+                      llvm::cl::init(false));
+
 bool needToPackSubByteElementBitWidth(unsigned bitWidth) {
+  // Enable i1 support if requested.
+  if (clEnableI1Support) {
+    return bitWidth == 1;
+  }
```

Comment:
Shouldn't this be something like below? Otherwise, i2/i4 won't be packed back-to-back when we enable the feature?

```suggestion
  // Enable i1 support if requested.
  if (clEnableI1Support && bitWidth == 1) {
    return true;
  }
```

Reply:
You are right!